my-app/src/Pieces/King.tsx

```diff
@@ -4,16 +4,37 @@ import  { ChessKing } from '@emotion-icons/fa-solid'
 import { Pieces } from "../Enums/PieceEnum";
 
 export class King extends Piece{ 
-   hasMoved: boolean
+
+   static whiteCastleLeftMove = new Coordinate(7,2);
+   static whiteCastleRightMove = new Coordinate(7,6);
+   static blackCastleLeftMove = new Coordinate(0,2);
+   static blackCastleRightMove = new Coordinate(0,6);
+   static possibleWhiteCastleMoves = [King.whiteCastleLeftMove, King.whiteCastleRightMove];
+   static possibleBlackCastleMoves = [King.blackCastleLeftMove, King.blackCastleRightMove];
 
    constructor(coordinate: Coordinate, color: string) {
       super(ChessKing, coordinate, color, Pieces.KING);
-      this.hasMoved = false;
    }
    
    isValidMove( moveLocation: Coordinate): boolean {
-      return [this.currentLocation.x + 1, this.currentLocation.x - 1, this.currentLocation.x].includes(moveLocation.x)
+      let isNormalMove = [this.currentLocation.x + 1, this.currentLocation.x - 1, this.currentLocation.x].includes(moveLocation.x)
       && [this.currentLocation.y + 1, this.currentLocation.y - 1, this.currentLocation.y].includes(moveLocation.y)
       && !moveLocation.equals(this.currentLocation);
```

Comment:
might make since to just put this logic in a method like the isCastlingMove and just return
```
return this.isNormalMove(moveLocation) || this.isCastlingMove
```

Reply:
I left it as is, since I didn't have a use for a isNormalMove function at the moment